In [1]:
import pandas as pd
import sys
sys.path.append("../")
from functools import reduce
from src.haversine import haversine

Vamos a utilizar los datas que hemos obtenido anteriomente.

In [2]:
companies = pd.read_csv('./data/companies.csv')

In [3]:
data_companies = pd.DataFrame(companies)
data_companies

Unnamed: 0            name  longitude   latitude
0           0             Fon  -3.632687  40.530724
1           1         Spotify   6.121511  49.607295
2           2      Videoplaza  -0.102089  51.521248
3           3  Veeam Software -83.088566  40.088052
4           4           Busuu  -3.683032  40.422549
5           6       Comunitae  -3.703340  40.490511
6           7          IMASTE  -3.675517  40.427245
7           8        ticketea  -3.706176  40.445516

In [4]:
school = pd.read_csv('./data/school.csv')

In [5]:
data_school = pd.DataFrame(school)
data_school.head()

Unnamed: 0                                       name    latitud  longitud  \
0           0      Escuela infantil municipal Agua Dulce  40.454061 -3.709371   
1           1          Escuela infantil municipal Altair  40.378132 -3.768701   
2           2        Escuela infantil municipal Amanecer  40.401990 -3.601044   
3           3  Escuela infantil municipal Ana de Austria  40.495522 -3.655737   
4           4      Escuela infantil municipal Campanilla  40.452700 -3.701998   

                                       geometry  
0  POINT (-3.709371361756636 40.45406088024461)  
1  POINT (-3.768700549230208 40.37813189249392)  
2  POINT (-3.601044345956354 40.40198995600377)  
3  POINT (-3.655736850911649 40.49552153813781)  
4    POINT (-3.701998076478461 40.452700492043)

In [6]:
metro = pd.read_csv('./data/metro.csv')

In [7]:
data_metro = pd.DataFrame(metro)


In [8]:
starbucks = pd.read_csv('./data/starbucks.csv')

In [9]:
data_starbucks = pd.DataFrame(starbucks)
data_starbucks.head()

Unnamed: 0       name    latitud  longitud  \
0           0  Starbucks  40.426067 -3.675617   
1           1  Starbucks  40.432968 -3.683220   
2           2  Starbucks  40.433564 -3.686829   
3           3  Starbucks  40.424975 -3.684107   
4           4  Starbucks  40.425760 -3.691782   

                                       geometry  
0  POINT (-3.675616746905888 40.42606680113607)  
1  POINT (-3.683220148086547 40.43296760669224)  
2   POINT (-3.686829359536097 40.4335639178076)  
3    POINT (-3.684107212216363 40.424974636524)  
4  POINT (-3.691781759262085 40.42576035726452)

### Con la función haversine obtenemos las empresas que tiene un colegio a menos de un `1000 mentros`

In [10]:
data = data_companies
data_school 
score_school = []

for elem in data.index:
    cord=[]
    cord.append(data.longitude[elem])
    cord.append(data.latitude[elem])
    for cole in data_school.index:
        cord_c = []
        cord_c.append(data_school.longitud[cole])
        cord_c.append(data_school.latitud[cole])
        distance = haversine(cord,cord_c)
        
        if distance <= 1000:
            score_school.append(data.name[elem])
        
print(score_school)

['IMASTE', 'ticketea', 'ticketea']


 ### Ahora buscamos las empresas que tienen Starbucks a menos de `500 metros`.

In [11]:
data = data_companies
 
score_starbucks = []

for elem in data.index:
    cord=[]
    cord.append(data.longitude[elem])
    cord.append(data.latitude[elem])
    for s in data_starbucks.index:
        cord_c = []
        cord_c.append(data_starbucks.longitud[s])
        cord_c.append(data_starbucks.latitud[s])
        distance = haversine(cord,cord_c)
        
        if distance <= 500:
            score_starbucks.append(data.name[elem])
        
print(score_starbucks)

['Busuu', 'Busuu', 'IMASTE']


### En el último paso obtenemos las empresas que tiene  estaciones de metros a menos de '300 metros'.

In [12]:
data = data_companies
 
score_metro = []

for elem in data.index:
    cord=[]
    cord.append(data.longitude[elem])
    cord.append(data.latitude[elem])
    for s in data_metro.index:
        cord_c = []
        cord_c.append(data_metro.longitud[s])
        cord_c.append(data_metro.latitud[s])
        distance = haversine(cord,cord_c)
        
        if distance <= 300:
            score_metro.append(data.name[elem])
        
print(score_metro)

['Busuu', 'IMASTE']


In [13]:
score_school, score_starbucks,score_metro

(['IMASTE', 'ticketea', 'ticketea'],
 ['Busuu', 'Busuu', 'IMASTE'],
 ['Busuu', 'IMASTE'])

## Conclusión: Nuestra mejor ubicación es  (-3.675517  40.427245), de la empresa IMASTE

In [14]:
print(companies[companies.name == 'IMASTE'])

   Unnamed: 0    name  longitude   latitude
6           7  IMASTE  -3.675517  40.427245
